iothub/service/tests/Exceptions/IotHubServiceExceptionTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.Tests.Exceptions
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class IotHubServiceExceptionTests
+    {
+        private const string Message = "sample message";
+
+        [TestMethod]
+        public void IotHubServiceException_Ctor_WithNonTransientErrorCode_Ok()
```

Comment:
Do you want to consider data-driven tests for these? It seems what changes is the status code and IoT Hub service error code, and what is validated is both of those values and whether it is transient.

Reply:
Sure!